### PR TITLE
feat(ui-react-native): Add theming for TextField primitive

### PR DESCRIPTION
**Pull request by joebuono**
#### Description of changes

- Add theming
- Update unit tests

#### Description of how you validated changes

Ran in iOS and Android. 

#### Checklist

- [X] PR description included
- [X] `yarn test` passes
- [X] Tests are updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-native/src/primitives/TextField/styles.ts [hunk 1]
```diff
@@ -1,27 +1,36 @@
...
+  return StyleSheet.create({
+    container: {
+      alignItems: 'flex-start',
+      ...components?.textfield.container,
+    },
+    inputContainer: {
+      alignItems: 'center',
+      borderColor: colors.black,
+      borderRadius: space.xs,
+      borderWidth: 1,
```
Comment (calebpollman) [inline packages/react-native/src/primitives/TextField/styles.ts:18]: Let's use a theme value here

File: packages/react-native/src/theme/types.ts [hunk 2]
```diff
@@ -44,6 +45,7 @@ export interface ComponentStyles {
   label: LabelStyles;
   radio: RadioStyles;
   radiogroup: RadioGroupStyles;
+  textfield: TextFieldStyles;
```
Comment (calebpollman) [inline packages/react-native/src/theme/types.ts:48]: Why are we using lowercase instead of camelCase for `textfield` (and `radiogroup`)?

Comment (joebuono) [reply packages/react-native/src/theme/types.ts:48]: I was just following [this pattern](https://github.com/aws-amplify/amplify-ui/blob/main/packages/ui/src/theme/tokens/components/index.ts#L93).

File: packages/react-native/src/primitives/TextField/styles.ts (hunk 1 above)
Comment (joebuono) [reply packages/react-native/src/primitives/TextField/styles.ts:18]: There are no tokens that resolve to 1.

**Review by ioanabrooks — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Pull request merged**